### ISSUE TITLE
Handling Wildcard and Op in same range

### ISSUE THREAD

**Issue by ScottAbbey** [steveklabnik/semver-parser#14]
When an operator is specified and a wildcard also exists, what should happen?

For example, if the range was `"< 4.2.*"`, it will turn into:

```
Predicate {
    op: Op::Wildcard(WildcardVersion::Patch),
    major: 4,
    minor: Some(2),
    patch: None
    pre: Vec::new(),
}
```

This is just throwing away the fact that the `<` (or any other operator) existed before the string.

My outside perspective is that such a version string should be rejected as nonsensical, but I couldn't find any discussion of this in the Cargo docs or the semver crate.

As far as I can tell, the behavior has remained the same both before and after the recent changes that removed the dependency on regex.

**Comment by steveklabnik** [comment]
Yeah, this is very strange. Hm.

**Comment by steveklabnik** [comment]
I want to say "reject as nonsense", but npm supports this, and treats it as `< 4.2.0`, basically. Which *is* sensible, I guess. So we should probably fix that.

**Comment by dtolnay** [comment]
This is fixed in semver 1.0.0 &mdash; `< 4.2.*` is parsed identically to `< 4.2`.

**Comment by steveklabnik** [comment]
This crate is no longer being used or maintained: https://github.com/steveklabnik/semver-parser/commit/dc6c1077899da46e89242041dda54ca36c170982
